src/librustc_mir/transform/const_prop.rs

```diff
@@ -783,10 +778,19 @@ impl CanConstProp {
             // FIXME(oli-obk): lint variables until they are used in a condition
             // FIXME(oli-obk): lint if return value is constant
             let local_kind = body.local_kind(local);
-            cpv.local_kinds.insert(local, clone_local_kind(&local_kind));
-            if local_kind == LocalKind::Arg {
+            cpv.local_kinds.push(local_kind);
```

Comment:
You can use this as the initializer closure for `IndexVec::from_fn_n()` and then instead of doing `body.local_kind(local)` here, you can just do `cpv.local_kinds[local]`.

Reply:
I did it and it seems to work! :)